azurelinuxagent/common/cgroup.py

```diff
@@ -13,21 +13,60 @@
 # limitations under the License.
 #
 # Requires Python 2.6+ and Openssl 1.0+
-from collections import namedtuple
 
 import errno
 import os
 import re
+from datetime import timedelta
 
-from azurelinuxagent.common import logger
+from azurelinuxagent.common import logger, conf
 from azurelinuxagent.common.exception import CGroupsException
 from azurelinuxagent.common.future import ustr
 from azurelinuxagent.common.osutil import get_osutil
 from azurelinuxagent.common.utils import fileutil
 
+REPORT_EVERY_HOUR = timedelta(hours=1)
+DEFAULT_REPORT_PERIOD = timedelta(seconds=conf.get_cgroup_check_period())
+
 AGENT_NAME_TELEMETRY = "walinuxagent.service"  # Name used for telemetry; it needs to be consistent even if the name of the service changes
 
-MetricValue = namedtuple('Metric', ['category', 'counter', 'instance', 'value'])
+
+class CounterNotFound(Exception):
+    pass
+
+
+class MetricValue(object):
+
+    """
+    Class for defining all the required metric fields to send telemetry.
+    """
+
+    def __init__(self, category, counter, instance, value, report_period=DEFAULT_REPORT_PERIOD):
+        self.__category = category
```

Comment:
These would probably be better as starting with a single underscore, not two. See [this stack overflow question](https://stackoverflow.com/questions/1301346/what-is-the-meaning-of-single-and-double-underscore-before-an-object-name) and its answers for some interesting discussion.